spock-core/src/main/java/org/spockframework/mock/runtime/MockInteraction.java

```diff
@@ -98,6 +99,26 @@ public int computeSimilarityScore(IMockInvocation invocation) {
     return score;
   }
 
+  @Override
+  public String describeMismatch(IMockInvocation invocation) {
+    List<String> mismatches = new ArrayList<>();
+    for (IInvocationConstraint constraint : constraints) {
+      boolean satisfied;
+      try {
+        satisfied = constraint.isSatisfiedBy(invocation);
+      } catch (Exception e) {
+        // can happen because we evaluate (code) argument constraints
+        // even if target and/or method constraints aren't satisfied
+        satisfied = false;
+      }
+
+      if (!satisfied) {
```

Comment:
Couldn't it be put into a try..catch block?

```
      try {
        if (!constraint.isSatisfiedBy(invocation)) {
          mismatches.add(constraint.describeMismatch(invocation));
        }
      } catch (Exception e) {
        // can happen because we evaluate (code) argument constraints
        // even if target and/or method constraints aren't satisfied
      }
```